src/scripts/main.js

```diff
@@ -1 +1,84 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+function creatDiv(str, className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.classList.add(className);
+
+  div.textContent = str;
+  body.appendChild(div);
+}
+
+function firstPromise() {
+  return new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      reject('First promise was rejected');
+    }, 3000);
+  });
+}
+
+function secondPromise() {
+  return new Promise((resolve) => {
+    document.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+
+      resolve('Second promise was resolved');
+    });
+  });
+}
+
+function thirdPromise() {
+  return new Promise((resolve) => {
+    document.addEventListener('click', () => {
+      document.addEventListener('contextmenu', (e) => {
+        e.preventDefault();
+        resolve('Third promise was resolved');
+      });
+    });
+
+    document.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+
+      document.addEventListener('click', () => {
+        resolve('Third promise was resolved');
+      });
```

Comment:
The nested event listeners in `thirdPromise` can lead to multiple attachments of the same event listener, which might cause unexpected behavior. Consider restructuring the logic to avoid this.